src/WebJobs.Script.Host/Program.cs

```diff
@@ -22,8 +26,35 @@ static void Main(string[] args)
                 RootScriptPath = rootPath
             };
 
+            // For local execution, allow reading secrets from a file. 
+            // This is an optional place to get secrets, useful in local testing. 
+            string secretsFile = Path.Combine(rootPath, "secrets.json");
+            if (File.Exists(secretsFile))
+            {
+                string json = File.ReadAllText(secretsFile);
+                var dict = JsonConvert.DeserializeObject<IDictionary<string, string>>(json);
+                config.AppSettings = new FileSecretNameResolver(dict);
```

Comment:
I really don't think an AppSettings bag belongs on ScriptHostConfiguration. ScriptHostConfig.HostConfig already has an INameResolver that we default to our own NameResolver instance (the class name is NameResolver) that provides some behavior. I think you should add your DefaultAppSettingResolver logic to that one so the logic is available to all the bindings that are already getting that resolver instance.

Also, I hope we can avoid having this secrets.json file pattern. It just opens up the door for people to accidentally check that file in. That's why we use env vars for local debug.